### PR TITLE
Add error code to duplicate output files

### PR DESCRIPTION
**Pull request by hellosnow**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs [hunk 1]
```diff
@@ -26,8 +27,12 @@ public static void RemoveDuplicateOutputFiles(ManifestItemCollection manifestIte
                               .GroupBy(obj => obj.relativePath, FilePathComparer.OSPlatformSensitiveStringComparer)
                               .Where(g => g.Count() > 1))
             {
-                Logger.LogWarning($"Overwrite occurs while input files \"{string.Join(", ", duplicates.Select(duplicate => duplicate.item.SourceRelativePath))}\" writing to the same output file \"{duplicates.Key}\"");
-                itemsToRemove.UnionWith(duplicates.Skip(1).Select(duplicate => duplicate.item.SourceRelativePath));
+                var duplicateSources = duplicates.Skip(1).Select(duplicate => duplicate.item.SourceRelativePath).ToList();
+                // TODO: plan to change this warning to error, add error code to analyze the impact.
+                Logger.LogWarning(
+                    $"Multiple input files are attempting to write to the same output file \"{duplicates.Key}\". Input file \"{duplicates.First().relativePath}\" is selected as the output content, please rename other input files to avoid duplicate output files: \"{string.Join(", ", duplicateSources)}\".",
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs]: >Input file \"{duplicates.First().relativePath}\" is selected as the output content [](start = 113, length = 82)

It's not sure which one is selected as they are built parallelly

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs]: Yes you're right, I'll update the msg.

---
In reply to: [112098532](https://github.com/dotnet/docfx/pull/1561#discussion_r112098532) [](ancestors = 112098532)

File: src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs [hunk 2]
```diff
@@ -26,8 +27,12 @@ public static void RemoveDuplicateOutputFiles(ManifestItemCollection manifestIte
                               .GroupBy(obj => obj.relativePath, FilePathComparer.OSPlatformSensitiveStringComparer)
                               .Where(g => g.Count() > 1))
             {
-                Logger.LogWarning($"Overwrite occurs while input files \"{string.Join(", ", duplicates.Select(duplicate => duplicate.item.SourceRelativePath))}\" writing to the same output file \"{duplicates.Key}\"");
-                itemsToRemove.UnionWith(duplicates.Skip(1).Select(duplicate => duplicate.item.SourceRelativePath));
+                var duplicateSources = duplicates.Skip(1).Select(duplicate => duplicate.item.SourceRelativePath).ToList();
+                // TODO: plan to change this warning to error, add error code to analyze the impact.
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs]: what is the plan? even if it is an error, we should fail after all documents are built?

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs]: offline discussed, yes should log error and throw after all docs are built.

---
In reply to: [112105890](https://github.com/dotnet/docfx/pull/1561#discussion_r112105890) [](ancestors = 112105890)

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**